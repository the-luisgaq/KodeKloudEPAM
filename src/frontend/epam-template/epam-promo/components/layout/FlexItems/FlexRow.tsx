import * as types from '../../types';
import { withMods } from '@epam/uui-core';
import * as uui from '@epam/uui';

interface FlexRowMods extends types.RowSizeMod {
    /**
     * @default 'none'
     */
    background?: 'white' | 'gray5' | 'none' | uui.FlexRowProps['background'];
}
export type FlexRowProps = Omit<uui.FlexRowProps, 'background'> & FlexRowMods;

export const FlexRow = withMods<Omit<uui.FlexRowProps, 'background'>, FlexRowProps>(uui.FlexRow, (props) => {
    return [`uui-flex-row-bg-${props.background || 'none'}`];
});
