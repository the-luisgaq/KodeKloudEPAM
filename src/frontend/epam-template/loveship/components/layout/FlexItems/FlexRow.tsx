import * as uuiCore from '@epam/uui-core';
import * as uui from '@epam/uui';

export interface RowMods extends Omit<uui.FlexRowMods, 'spacing' | 'background'> {
    /**
     * @default 'none'
     */
    background?: 'white' | 'night50' | 'night100' | 'none';
    spacing?: '6' | '12' | '18' | null;
    /**
     * @default 'panel'
     */
    type?: 'form' | 'panel';
}

export type FlexRowProps = uuiCore.FlexRowProps & RowMods;

const commonDefaults: FlexRowProps = {
    size: '36',
    background: 'none',
};

const rowTypesDefaults: Record<string, FlexRowProps> = {
    form: {
        ...commonDefaults,
        columnGap: '12',
        alignItems: 'top',
        vPadding: '18',
        padding: '24',
    },
    panel: {
        ...commonDefaults,
        alignItems: 'center',
        columnGap: '6',
        size: '36',
    },
};

export const FlexRow = uuiCore.withMods<uuiCore.FlexRowProps, FlexRowProps>(
    uui.FlexRow,
    (props) => {
        return [`uui-flex-row-bg-${props.background || 'none'}`];
    },
    (props) => {
        const defaults = rowTypesDefaults[props.type || 'panel'];
        return { ...defaults, ...props };
    },
);
