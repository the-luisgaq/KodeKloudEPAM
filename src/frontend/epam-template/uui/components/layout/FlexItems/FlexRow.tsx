import React, { forwardRef } from 'react';
import {
    devLogger,
    FlexRowProps as uuiFlexRowProps,
    isEventTargetInsideClickable,
    uuiMarkers,
    cx, Overwrite,
} from '@epam/uui-core';

import { settings } from '../../../settings';

import css from './FlexRow.module.scss';

export interface FlexRowMods {
    /**
     *  Defines row size
     *  @default '36'
     */
    size?: null | '24' | '30' | '36' | '42' | '48';
    /** Pass true, to enable row top border */
    borderTop?: boolean;
    /** Pass true, to enable row bottom border */
    borderBottom?: boolean;
    /** Defines row margin */
    margin?: '12' | '24';
    /** Defines horizontal row padding */
    padding?: '6' | '12' | '18' | '24';
    /**
     * @deprecated
     * The `spacing` property is deprecated and will be removed in future versions. Please use `columnGap` instead.
     * Defines row spacing
     * */
    spacing?: '6' | '12' | '18';
    /** Pass true, to show a top shadow */
    topShadow?: boolean;
    /** Defines vertical row padding */
    vPadding?: '12' | '18' | '24' | '36' | '48';
    /** Flexbox column gap property [Flexbox Guide](https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-gap-row-gap-column-gap) */
    columnGap?: number | '6' | '12' | '18' | '24' | '36';
    /** Flexbox row gap property [Flexbox Guide](https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-gap-row-gap-column-gap) */
    rowGap?: number | '6' | '12' | '18' | '24' | '36';
    /** Defines row background */
    background?: 'surface-main';
}

export interface FlexRowModsOverride {}

/** Represents the properties of the FlexRow component. */
export interface FlexRowProps extends uuiFlexRowProps, Overwrite<FlexRowMods, FlexRowModsOverride> {}

export const FlexRow = forwardRef<HTMLDivElement, FlexRowProps>((props, ref) => {
    if (__DEV__) {
        if (props.spacing) {
            devLogger.warn('[FlexRow]: The `spacing` property is deprecated and will be removed in future versions. Please use `columnGap` instead.');
        }
    }

    const classNames = cx([
        css.root,
        'uui-flex-row',
        props.onClick && uuiMarkers.clickable,
        props.cx,
        props.size !== null && 'uui-size-' + (props.size || settings.flexRow.sizes.default),
        props.alignItems && css.alignItems,
        props.justifyContent && css.justifyContent,
        props.padding && css.padding,
        props.vPadding && css.vPadding,
        props.margin && css.margin,
        props.topShadow && css.topShadow,
        props.borderBottom && css.borderBottom,
        props.borderTop && css.borderTop,
        props.columnGap && css.columnGap,
        props.rowGap && css.rowGap,
        props.spacing && css.spacing,
        props.background && `uui-flex-row-bg-${props.background}`,
    ]);

    let alignItemsValue = props.alignItems;
    if (alignItemsValue === 'top') {
        alignItemsValue = 'flex-start';
    } else if (alignItemsValue === 'bottom') {
        alignItemsValue = 'flex-end';
    }

    const style = {
        ...(props.alignItems && { '--uui-flex-row-align-items': alignItemsValue }),
        ...(props.justifyContent && { '--uui-flex-row-justify-content': props.justifyContent }),
        ...(props.columnGap && { '--uui-flex-row-column-gap': `${props.columnGap}px` }),
        ...(props.rowGap && { '--uui-flex-row-row-gap': `${props.rowGap}px` }),
        ...(props.padding && { '--uui-flex-row-padding': `${props.padding}px` }),
        ...(props.vPadding && { '--uui-flex-row-v-padding': `${props.vPadding}px` }),
        ...(props.margin && { '--uui-flex-row-margin': `${props.margin}px` }),
        ...(props.spacing && { '--uui-flex-row-spacing': `${props.spacing}px` }),
    };

    return (
        <div
            ref={ ref }
            onClick={ props.onClick ? (e) => !isEventTargetInsideClickable(e) && props.onClick(e) : undefined }
            className={ classNames }
            { ...props.rawProps }
            style={ {
                ...style,
                ...props.rawProps?.style,
            } }
        >
            {props.children}
        </div>
    );
});
