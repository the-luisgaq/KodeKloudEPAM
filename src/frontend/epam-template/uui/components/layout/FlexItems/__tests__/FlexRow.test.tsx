import React from 'react';
import { FlexRow } from '../FlexRow';
import { render } from '@epam/uui-test-utils';

describe('FlexRow', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(<FlexRow />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const { asFragment } = render(<FlexRow
            onClick={ () => {} }
            margin="12"
            size="24"
            padding="12"
            borderBottom
            alignItems="top"
            topShadow
            vPadding="12"
            rowGap="24"
            columnGap="24"
            borderTop={ true }
            justifyContent="center"
            cx="some class"
            rawProps={ { id: '123' } }
            key="key"
            background="surface-main"
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});
