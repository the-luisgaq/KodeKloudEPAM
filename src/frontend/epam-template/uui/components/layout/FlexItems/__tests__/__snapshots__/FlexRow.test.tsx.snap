// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexRow should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="root uui-flex-row uui-size-36"
  />
</DocumentFragment>
`;

exports[`FlexRow should be rendered correctly with props 1`] = `
<DocumentFragment>
  <div
    class="root uui-flex-row -clickable some class uui-size-24 alignItems justifyContent padding vPadding margin topShadow borderBottom borderTop columnGap rowGap uui-flex-row-bg-surface-main"
    id="123"
    style="--uui-flex-row-align-items: flex-start; --uui-flex-row-justify-content: center; --uui-flex-row-column-gap: 24px; --uui-flex-row-row-gap: 24px; --uui-flex-row-padding: 12px; --uui-flex-row-v-padding: 12px; --uui-flex-row-margin: 12px;"
  />
</DocumentFragment>
`;
